### DISCUSSION TITLE
Wrong port while serving behind proxy

### DISCUSSION THREAD

**Discussion by miIiano** [firefly-iii/firefly-iii#5572] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

While behind docker proxy let's say 1.2.3.4 internal container ip serving on port 8080.

Cannot login as base url is set as https://domain:8080, js not fetched.

Refused to set the document's base URI to 'https://domain:8080/' because it violates the following Content Security Policy directive: "base-uri 'self'".  

     - DB_HOST=db.intranet
     - DB_PORT=3306
     - DB_CONNECTION=mysql
     - DB_DATABASE=fin
     - DB_USERNAME=fin
     - DB_PASSWORD=fin
     - APP_ENV=heroku   // to force set base https scheme
     - APP_KEY=###
     - APP_URL=https://domain
     - TRUSTED_PROXIES=**
     - USE_PROXIES=127.0.0.1

### Expected behaviour

follow https scheme port  

### Debug information

Container running under docker local port and exposed with nginx proxy manager
```
<!DOCTYPE html>
<html lang="en">
<head>
    <base href="https://domain:8080/">
    <meta charset="UTF-8">
    <meta name="robots" content="noindex, nofollow, noarchive, noodp, NoImageIndex, noydir">
    <meta name="apple-mobile-web-app-capable" content="yes">
```

**Comment by JC5** [comment]
If you need to set `APP_ENV=heroku` to make it use https, and the port is not working, your reverse proxy configuration is wrong. You'll need to set some headers, like so:

```
		proxy_set_header X-Forwarded-Host $host;
		proxy_set_header X-Forwarded-Server $host;
		proxy_set_header X-Forwarded-Proto $scheme;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

**Reply by miIiano** [reply]
It's already included. I checked already existing issues.

```
server {
  set $forward_scheme http;
  set $server         "172.110.1.13";
  set $port           8080;

  listen 80;
listen [::]:80;

listen 443 ssl http2;
listen [::]:443 ssl http2;

  server_name domain;

  # Let's Encrypt SSL
  include conf.d/include/letsencrypt-acme-challenge.conf;
  include conf.d/include/ssl-ciphers.conf;
  ssl_certificate /etc/letsencrypt/live/npm-3/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/npm-3/privkey.pem;
  include conf.d/include/assets.conf;
  include conf.d/include/block-exploits.conf;
  add_header Strict-Transport-Security "max-age=63072000;includeSubDomains; preload" always;
  include conf.d/include/force-ssl.conf;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection $http_connection;
proxy_http_version 1.1;
  access_log /data/logs/proxy-host-20_access.log proxy;
  error_log /data/logs/proxy-host-20_error.log warn;
proxy_set_header Host $http_host;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
proxy_set_header X-Forwarded-Proto $scheme;

  location / {
  add_header Strict-Transport-Security "max-age=63072000;includeSubDomains; preload" always;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_http_version 1.1;
    include conf.d/include/proxy.conf;
  }
  
  include /data/nginx/custom/server_proxy[.]conf;
}
```

**Reply by miIiano** [reply]
proxy_set_header X-Forwarded-Host $host;
		proxy_set_header X-Forwarded-Server $host;

still same with these

**Reply by ghost** [reply]
For what its worth my issue is identical as mentioned in my discussion. This is a much more thorough description of the issue.

**Reply by miIiano** [reply]
@JC5, should be marked as a issue? since following existing guides aren't working and moreover I'm not the only one with this issue

**Reply by JC5** [reply]
Sorry no. It works for hundreds of users but it doesn't work for 2. I'm going to have to assume the issue isn't with me.

I'm not sure what you expect from me at this point. I'm afraid I can't help you debug this remotely. I see you have variables you don't use, and config files I can't validate.

I suggest you go back to the basics; the lines I've shared are the basics that work for most people. More complex use cases aren't something I can debug for you.

**Reply by ghost** [reply]
Hey I think we might be able to round it to an issue with nginx-proxy-manager. Which flavor are you using Miliano? I am using jlesage's version and updated it as well when I updated firefly-iii. The only strange thing is everything else I am proxying still works just fine. Perhaps its some combination that is throwing it off? Checking similar issues for other services online I see springboard had an issue similar but the solution offered that worked was an nginx proxy location rewrite for every url pull which seems too blunt of an instrument.

**Reply by miIiano** [reply]
@MaroonedAnchor version doesn't matter, unless if it's really old, I'm using though an arm64 updated version which seems to keep track with everything till now with firefly. Many other complex configurations and systems are up and running just fine.

**Reply by ghost** [reply]
@miIiano Good News! The issue must have been related to Nginx-Proxy-Manager, new image just came out and after updating my docker instance Firefly-iii works just perfectly. Have not gone into the weeds but this was the an issue on the proxy level. I recommend you update and see if that works for you!

**Comment by miIiano** [comment]
Yeah... @MaroonedAnchor also for me is fine after updating. So all good for now.
